Code/Framework/AzToolsFramework/AzToolsFramework/AssetDatabase/AssetDatabaseConnection.cpp

```diff
@@ -2929,6 +2947,52 @@ namespace AzToolsFramework
                 return GetResult(callName, statement, handler);
             }
 
+            bool GetWildcardSourceDependencyResult(const char* callName, SQLite::Statement* statement, AssetDatabaseConnection::sourceFileDependencyHandler handler)
+            {
+                AZ_UNUSED(callName); // AZ_Error may be compiled out entirely in release builds.
+
+                Statement::SqlStatus result = statement->Step();
+
+                SourceFileDependencyEntry entry;
+                int absolutePathMatch;
+                auto boundColumns = CombineColumns(entry.GetColumns(), MakeColumns(MakeColumn("AbsolutePath", absolutePathMatch)));
+
+                bool validResult = result == Statement::SqlDone;
+                while (result == Statement::SqlOK)
+                {
+                    if (!boundColumns.Fetch(statement))
+                    {
+                        return false;
+                    }
+
+                    // Its possible for certain relative wildcard dependencies to match an absolute path (ex c%.foo matches c:/bar.foo)
+                    // If that happens, skip this result and move on to the next one
+                    if(!entry.m_dependsOnSource.IsUuid() && absolutePathMatch == 1 && !AZ::IO::PathView(entry.m_dependsOnSource.GetPath()).IsAbsolute())
+                    {
+                        result = statement->Step();
+                        continue;
+                    }
+
+                    if (handler(entry))
+                    {
+                        result = statement->Step();
+                    }
+                    else
+                    {
+                        result = Statement::SqlDone;
+                    }
+                    validResult = true;
+                }
+
+                if (result == Statement::SqlError)
+                {
+                    AZ_Warning(LOG_NAME, false, "Error occurred while stepping %s", callName);
```

Comment:
Nit: Maybe use AZ_Error instead of AZ_Warning here since the log message indicates that it's an error.